2015/presentations/Business logic with PostgreSQL and Python/text.md

```diff
@@ -0,0 +1,144 @@
+
+## Why not ORM?
+
+Using ORM as a business logic container definitely has a lot of pros, although there is one serious problem with it. If you have a project which uses DB that is shared with other projects and those use different languages...then... You are going to use API for such a interactions with third-party software projects. How about if you have to deal with legacy code where there is not such an option to have API? Where to have business logic with one source of truth. How about using DB as a main source of trust?
+
+## Python
+
+Compiling Python is not a difficult process. Let me show you in few simples steps how to do it.
+
+Download python source code
+
+    wget https://www.python.org/ftp/python/2.7.10/Python-2.7.10.tgz
+
+Compile it. Warning, Please pay attention about –enable-shared flag!
+
+    ~/stuff/Python-2.7.10% ./configure —PREFIX=/opt/py –enable-shared
+
+Now it is very important to remember to compile python with **–enable-shared** option. Without it during PostgreSQL compilation you're not going to be able to use compiled Python as procedural language extension.
+
+Also on some systems if you try to run Python and you’re getting this kind of error
+
+    py ➤ bin/python2.7
+
+**bin/python2.7: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory**
+
+Then you need to add lib path of your newly compiled python to the system lib path
+
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/py/lib
+    
+## PostgreSQL with Python support
+
+Download source code.
+
+    wget https://ftp.postgresql.org/pub/source/v9.4.4/postgresql-9.4.4.tar.bz2
+
+Compile with Python support from */opt/py* directory
+
+    ./configure –prefix=/opt/pgsql —with-python PYTHON=/opt/py/bin/python
+    make
+    make install
+
+Now compile and install PostgreSQL plugins
+
+    cd ~/stuff/postgresql-9.4.3/contrib
+    make
+    make install
+
+## Create DB
+
+Now it is time to start PostgreSQL. First you have to initialize name space.
+
+    mkdir /opt/pg_data
+    /opt/pgsql/bin/initdb /opt/pg_data
+
+and start DB…
+
+    /opt/pgsql/bin/postmaster -p 5432 -D /opt/pg_data/
+
+Create new DB.
+
+    /opt/pgsql/bin/createdb -h localhost  -E utf8 pie
+
+Please remember about the correct encoding for DB! Now create new language for database pie
+
+    /opt/pgsql/bin/createlang -h localhost  -d pie plpythonu
+    
+And that's it. Simple as that. Your new DB has full Python support and we can start orginizing business logic in there.
+
+## First function
+
+Before you are going to create your first plPython (PostgreSQL stored Python functions) you have to know how it works. Your newly compiled Python and its modules are fully accessible from plPython. That means that all the big library of Python modules is fully open for your access when you're going to write your business logic.

```

Comment:
That means that the entire Python standard library is fully accessible when writing your business logic.